hera_cal/lstbin_simple.py

```diff
@@ -947,12 +1099,78 @@ def lst_bin_files_single_outfile(
+    ignored for the averaging. In this inpainted mode, the Nsamples are the number of
+    un-flagged integrations (whether they were in-painted or not). The LST-binned flags
+    are only set to True if ALL of the nights for a given bl-channel-pol are flagged
+    (again, whether they were in-painted or not). In flagged mode, both the Nsamples
+    and Flags are the same as inpainted mode. The averaged data, however, ignores any
+    flagged data. This can lead to less spectrally-smooth results. The "STD" files 
+    contain LST-binned "data" that is the standard deviation of the data in each LST-bin.
+    This differs between inpainted and flagged modes in the same way as the "LST" files:
+    in inpainted mode, the flagged and inpainted data is used for calculating the sample
+    variance, while in flagged mode, only the unflagged data is used. The final flags
+    in the "STD" files is equivalent to that in the "LST" files. The Nsamples in the 
+    "STD" files is actually the number of unflagged nights in the LST bin (so, not the
+    sum of Nsamples), where "unflagged" really does mean unflagged -- whether inpainted
+    or not. 
+
+    One note here about what is considered "flagged" vs. "flagged and inpainted" vs
+    "flagged and not inpainted". In flagged mode, there are input flags that exist in the
+    input files. These are potentially *augmented* by sigma clipping within the LST
+    binner, and also by flagging whole LST bins if they have too few unflagged integrations.
+    In inpainted mode, input flags are considered as normal flags. However, only 
+    "non-inpainted" flags are *ignored* for the averaging. By default, all flagged data
+    is considered to to be in-painted UNLESS it is a blt-pol that is fully flagged (i.e.
+    all channels are flagged for an integration for a single bl and pol). However, you
+    can tell the routine that other data is NOT in-painted by supplying a "where inpainted"
+    file. Now, integrations in LST-bins that end up having "too few" unflagged 
+    integrations will be flagged inside the binner, however in inpainted mode, if these
+    are considered "inpainted", they will still be used in averaging (this means they
+    will have "valid" data for the average, but their average will be flagged). 
+    On the other hand, flags that are applied by sigma-clipping will be considered 
+    NOT inpainted, i.e. those data will be ignored in the averaged, just like flagging 
+    mode. In this case, either choice is bad: to include them in the average is bad
+    because even though they may have been actually in-painted, whatever value they have
+    is clearly triggering the sigma-clipper and is therefore an outlier. On the other
+    hand, to ignore them is bad because the point of in-painting mode is to get 
+    smoother spectra, and this negates that. So, it's best just to not do sigma-clipping
+    in inpainted mode.
```

Comment:
Can there be a mode where inpainted data cannot get sigma-clipped but non-inpainted data can?